Add KCSAN config

melver opened the pull request.

xairy commented:
Check out the KMSAN config generation [script](https://github.com/google/syzkaller/blob/master/dashboard/config/generate-config-kmsan-from-kasan.sh). I think it's better to do something similar for KCSAN.

dvyukov commented on dashboard/config/upstream-kcsan.config:
```diff
@@ -1,28 +1,33 @@
 
+# KCSAN enable
+CONFIG_KCSAN=y
+CONFIG_KCSAN_EARLY_ENABLE=n
```
Is `=n` even a thing? The standard way to disable something is `is not set`. make olddefconfig may fail to parse `=n`.

dvyukov commented:
I think we also want to disable these things:
https://github.com/google/syzkaller/blob/master/dashboard/config/bits-kmsan.config#L24-L44
These will slowdown execution and generate lots of crashes that will throw away anyway.

melver commented:
Done.

- Added script to generate KCSAN config.
- Added generated config.
- Removed intermediate commit, since it should be clear from generator script what the changes are.

dvyukov commented on dashboard/config/bits-kcsan.config:
```diff
@@ -0,0 +1,47 @@
+# Avoid reboot loop in instrumented kernel:
+# CONFIG_PREEMPT is not set
+CONFIG_PREEMPT_NONE=y
+# CONFIG_PARAVIRT is not set
```
We don't need this whole block. This is just a work around some KMSAN bugs.
CONFIG_PARAVIRT should work and it useful. syzbot is PARAVIRT.

dvyukov commented on dashboard/config/bits-kcsan.config:
```diff
@@ -0,0 +1,47 @@
+# CONFIG_PARAVIRT is not set
+
+# Disable debug configs enabled in KASAN configs:
+# CONFIG_BLK_DEBUG_FS is not set
```
No need to disable this one. It's just some entries in debugfs.

dvyukov commented on dashboard/config/bits-kcsan.config:
```diff
@@ -0,0 +1,47 @@
+# Disable debug configs enabled in KASAN configs:
+# CONFIG_BLK_DEBUG_FS is not set
+# CONFIG_DEBUG_DEVRES is not set
+# CONFIG_DYNAMIC_DEBUG is not set
```
This may be useful as it can find races on arguments to printk.

dvyukov commented:
Also build/boot this config in parallel, so that we know it works.

dvyukov commented:
LGTM

xairy commented:
LGTM

melver commented:
Done.

Tested with local syzkaller, which is still finding plenty of races.
